tests/behat/serve-bootstrap.php

```diff
@@ -37,11 +40,35 @@
 	define('SS_DATABASE_CHOOSE_NAME', true);
 }
 
+// Ensure Director::protocolAndHost() works
+if (empty($_SERVER['HTTP_HOST'])) {
+	$_SERVER['HTTP_HOST'] = 'localhost';
+}
+
+// Default database settings
+global $project;
+$project = 'mysite';
+
 // Default database settings
 global $database;
 $database = '';
 
-require_once(__DIR__ . '/../../conf/ConfigureFromEnv.php');
+// Mock installer 'mysite' ONLY if installed with 'cms'
+$projectPath = BASE_PATH.'/'.$project;
+if (is_dir(BASE_PATH .'/cms') && !is_dir($projectPath)) {
+	$pageBase = SiteTree::class;
+	$controllerBase = ContentController::class;
+	$pageDotPHP = <<<EOS
+<?php
+// Auto-generated via serve-bootstrap.php
+
+class Page extends $pageBase {}
```

Comment:
This might be better done as a .travis.yml step (e.g. a `cp` operation of a Page.php file from tests somewhere) so that it can be loaded into the manifest for behat as well as phpunit?